src/Microsoft.Azure.WebJobs.Host/Indexers/FunctionIndexer.cs

```diff
@@ -351,23 +343,72 @@ private bool HasJobAttribute(CustomAttributeData attributeData)
             };
         }
 
-        private class ListenerFactory : IListenerFactory
+        internal class ListenerFactory : IListenerFactory
         {
             private readonly FunctionDescriptor _descriptor;
             private readonly ITriggeredFunctionExecutor _executor;
             private readonly ITriggerBinding _binding;
+            private readonly TraceWriter _trace;
 
-            public ListenerFactory(FunctionDescriptor descriptor, ITriggeredFunctionExecutor executor, ITriggerBinding binding)
+            public ListenerFactory(FunctionDescriptor descriptor, ITriggeredFunctionExecutor executor, ITriggerBinding binding, TraceWriter trace)
             {
                 _descriptor = descriptor;
                 _executor = executor;
                 _binding = binding;
+                _trace = trace;
             }
 
-            public Task<IListener> CreateAsync(CancellationToken cancellationToken)
+            public async Task<IListener> CreateAsync(CancellationToken cancellationToken)
             {
                 ListenerFactoryContext context = new ListenerFactoryContext(_descriptor, _executor, cancellationToken);
-                return _binding.CreateListenerAsync(context);
+                IListener listener = await _binding.CreateListenerAsync(context);
+                return new TriggerListener(listener, _descriptor, _trace);
+            }
+
+            private class TriggerListener : IListener
+            {
+                private readonly IListener _listener;
+                private readonly FunctionDescriptor _descriptor;
+                private readonly TraceWriter _trace;
+
+                public TriggerListener(IListener listener, FunctionDescriptor descriptor, TraceWriter trace)
+                {
+                    _listener = listener;
+                    _descriptor = descriptor;
+                    _trace = trace;
+                }
+
+                public void Cancel()
+                {
+                    _listener.Cancel();
+                }
+
+                public void Dispose()
+                {
+                    _listener.Dispose();
+                }
+
+                public async Task StartAsync(CancellationToken cancellationToken)
+                {
+                    try
+                    {
+                        await _listener.StartAsync(cancellationToken);
+                    }
+                    catch (Exception e)
+                    {
+                        var fex = new FunctionIndexingException(_descriptor.ShortName, e);
+                        _trace.Error(e.Message, fex);
+                        if (!fex.Handled)
+                        {
+                            throw;
+                        }
+                    }
+                }
+
+                public Task StopAsync(CancellationToken cancellationToken)
+                {
```

Comment:
If you look at all our other listeners, I don't think we're very good about this in general. If we think we have improvements in this area to make to listeners, I'd say its a new bug. Traditionally from the core SDK point of view, listener startup exceptions are intended to bring down the entire host and process. In that case, stop would never be called.